server/routerlicious/packages/services/src/mongoExceptionRetryRules/mongoError/index.ts

```diff
@@ -227,6 +227,19 @@ class RequestTimedOutBulkWriteErrorRule extends BaseMongoExceptionRetryRule {
 	}
 }
 
+class ConnectionPoolClearedErrorRule extends BaseMongoExceptionRetryRule {
+	private static readonly errorName = "MongoPoolClearedError";
+	protected defaultRetryDecision: boolean = true;
+
+	constructor(retryRuleOverride: Map<string, boolean>) {
+		super("ConnectionPoolClearedErrorRule", retryRuleOverride);
+	}
+
+	public match(error: any): boolean {
+		return error.name && (error.name as string) === ConnectionPoolClearedErrorRule.errorName;
```

Comment:
Do we have a sample of this error?

Also are we sure this error is retryable?

Reply:
Hey @zhangxin511 , below are some examples:
1. `{"address"=>"frs-prod-cosmos-group-seasia-seasia-001-eastasia.mongo.cosmos.azure.com:10255", "name"=>"MongoPoolClearedError", "message"=>"Connection pool for frs-prod-cosmos-group-seasia-seasia-001-eastasia.mongo.cosmos.azure.com:10255 was cleared because another operation failed with: \"connection 1 to 20.205.82.1:10255 closed\"" ...`
2. `{"address"=>"frs-prod-cosmos-group-seasia-seasia-001-eastasia.mongo.cosmos.azure.com:10255", "name"=>"MongoPoolClearedError", "message"=>"Connection pool for frs-prod-cosmos-group-seasia-seasia-001-eastasia.mongo.cosmos.azure.com:10255 was cleared because another operation failed with: \"Client network socket disconnected before secure TLS connection was established\"" ...`

They seem like connection/network errors, so I made them retryable. I am open to changing that if you think they are not.